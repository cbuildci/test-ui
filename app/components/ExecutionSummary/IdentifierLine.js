/**
 * IdentifierLine
 */

import React from 'react';
import PropTypes from 'prop-types';

function IdentifierLine({
    githubHost,
    owner,
    repo,
    commit,
    executionNum,
}) {
    const urlPrefix = githubHost
        ? `https://${githubHost}/`
        : '/app/repo/';

    return (
        <span className="d-inline-flex flex-wrap">
            <span className="mr-2">
                <i className="fab fa-github fa-fw text-muted mr-1"/>
                <a href={`${urlPrefix}${owner}/${repo}`}>{owner}/{repo}</a>
            </span>

            <span className="mr-2">
                <i className="fas fa-code-branch fa-fw text-muted mr-1"/>
                <a href={`${urlPrefix}${owner}/${repo}/commit/${commit}`}>{commit.substr(0, 10)}</a>
            </span>

            <span className="mr-2">
                <i className="fas fa-shipping-fast fa-fw text-muted mr-1"/>
                #{executionNum}
            </span>
        </span>
    );
}

IdentifierLine.propTypes = {
    githubHost: PropTypes.string,
    owner: PropTypes.string.isRequired,
    repo: PropTypes.string.isRequired,
    commit: PropTypes.string.isRequired,
    executionNum: PropTypes.number.isRequired,
};

IdentifierLine.defaultProps = {
    githubHost: null,
};

export default IdentifierLine;
