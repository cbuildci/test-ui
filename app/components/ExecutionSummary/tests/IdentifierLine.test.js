import React from 'react';
import { shallow, render } from 'enzyme';
import { IntlProvider } from 'react-intl';
import IdentifierLine from '../IdentifierLine';

const snapshots = (jsx) => {
    expect(shallow(jsx)).toMatchSnapshot();
    expect(render(
        <IntlProvider locale="en" initialNow={1500000100000}>{jsx}</IntlProvider>
    )).toMatchSnapshot();
};

describe('<IdentifierLine/>', () => {
    it('should render expected JSX', () => {
        snapshots(
            <IdentifierLine
                githubHost="git.com"
                owner="foo"
                repo="bar"
                commit="edd98b1508f303eabaa12464f1dbb8ec5aace9de"
                executionNum={5}
            />
        );
    });
});
