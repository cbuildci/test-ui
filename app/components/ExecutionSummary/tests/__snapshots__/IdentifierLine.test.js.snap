// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IdentifierLine/> should render expected JSX 1`] = `
<span
  className="d-inline-flex flex-wrap"
>
  <span
    className="mr-2"
  >
    <i
      className="fab fa-github fa-fw text-muted mr-1"
    />
    <a
      href="https://git.com/foo/bar"
    >
      foo
      /
      bar
    </a>
  </span>
  <span
    className="mr-2"
  >
    <i
      className="fas fa-code-branch fa-fw text-muted mr-1"
    />
    <a
      href="https://git.com/foo/bar/commit/edd98b1508f303eabaa12464f1dbb8ec5aace9de"
    >
      edd98b1508
    </a>
  </span>
  <span
    className="mr-2"
  >
    <i
      className="fas fa-shipping-fast fa-fw text-muted mr-1"
    />
    #
    5
  </span>
</span>
`;

exports[`<IdentifierLine/> should render expected JSX 2`] = `
<span
  class="d-inline-flex flex-wrap"
>
  <span
    class="mr-2"
  >
    <i
      class="fab fa-github fa-fw text-muted mr-1"
    />
    <a
      href="https://git.com/foo/bar"
    >
      foo/bar
    </a>
  </span>
  <span
    class="mr-2"
  >
    <i
      class="fas fa-code-branch fa-fw text-muted mr-1"
    />
    <a
      href="https://git.com/foo/bar/commit/edd98b1508f303eabaa12464f1dbb8ec5aace9de"
    >
      edd98b1508
    </a>
  </span>
  <span
    class="mr-2"
  >
    <i
      class="fas fa-shipping-fast fa-fw text-muted mr-1"
    />
    #5
  </span>
</span>
`;
